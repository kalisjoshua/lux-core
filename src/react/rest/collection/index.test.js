import React from 'react';
import renderer from 'react-test-renderer';

import registry from '../../componentRegistry';

import './index';
const Collection = registry('Rest.Collection');

const CreateButton = registry('Rest.Collection.CreateButton');
const Paging = registry('Paging');
const RestCollectionList = registry('Rest.Collection.List');

const props = {
  actions: [
    {
      href: 'http://foo.bar/baz',
      name: 'create',
      title: 'Foo Bar Baz',
    }
  ],
  entities: [],
  links: [
    {
      href: '/',
      rel: ['self'],
      title: 'Resources'
    }
  ],
  properties: {},
  title: '',
};

describe('Collection', function () {
  afterEach(function () {
    registry('Rest.Collection.CreateButton', CreateButton);
    registry('Rest.Collection.List', RestCollectionList);
    registry('Paging', Paging);
  });

  it('should exist; and should be a function', function () {
    expect(typeof Collection).toBe('function');
  });

  it('should throw if no `self` link', function () {
    const attrs = {
      ...props,
    };
    attrs.links = [];

    expect(function () {
      renderer.create(<Collection {...attrs}/>)
    }).toThrow('No `self` link provided from API in response.');
  });

  it('should', function () {
    registry('Rest.Collection.CreateButton', () => (<p />));
    registry('Rest.Collection.List', () => (<p />));
    registry('Paging', () => (<p />));

    const attrs = {
      ...props,
    };

    const component = renderer.create(
      <Collection {...attrs} />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
