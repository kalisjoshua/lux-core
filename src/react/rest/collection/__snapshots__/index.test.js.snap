exports[`Collection should render a collection page; even if the \`actions\` are missing 1`] = `
<div
  className="collection-display">
  <h2
    className="heading--primary">
    Resources
  </h2>
  <p>
    No more 
    Resources
     to show.
  </p>
  <noscript />
</div>
`;

exports[`Collection should render a collection page; with no errors 1`] = `
<div
  className="collection-display">
  <h2
    className="heading--primary">
    Resources
  </h2>
  <a
    className="btn btn--primary mrl"
    href="http://foo.bar/baz/new"
    onClick={[Function]}
    title="Foo Bar Baz">
    Foo Bar Baz
  </a>
  <p>
    No more 
    Resources
     to show.
  </p>
  <noscript />
</div>
`;
